zerver/views/development/integrations.py

```diff
@@ -0,0 +1,54 @@
+import os
+import ujson
+from typing import Dict
+
+from django.http import HttpRequest, HttpResponse
+from django.shortcuts import render
+from django.utils.translation import ugettext as _
+
+from zerver.lib.integrations import WEBHOOK_INTEGRATIONS
+from zerver.lib.request import has_request_variables, REQ
+from zerver.lib.response import json_success, json_error
+
+
+ZULIP_PATH = os.path.join(os.path.dirname(os.path.abspath(__file__)), '../../../')
+
+
+def get_webhook_integrations():
+    return [integration.name for integration in WEBHOOK_INTEGRATIONS]
+
+
+def dev_panel(request: HttpRequest) -> HttpResponse:
+
+    if request.method == "GET":
+        integrations = get_webhook_integrations()
+        context = {"integrations": integrations}
+        return render(request, "zerver/integrations/development/dev_panel.html", context)
+
+    elif request.method == "POST":
+        return json_success({"msg": "success"})
+
+
+@has_request_variables
+def get_fixtures(request: HttpResponse, integration_name: REQ(str)) -> HttpResponse:
+    integrations = get_webhook_integrations()
+    if integration_name not in integrations:
+        return json_error(_("\"%s\" is not a valid webhook integration.") % (integration_name))
+
+    fixtures_pack = {}
+    fixtures_dir = os.path.join(ZULIP_PATH, "zerver/webhooks/%s/fixtures" % (integration_name))
+    if not os.path.exists(fixtures_dir):
+        if not os.path.exists(os.path.join(ZULIP_PATH, "zerver/webhooks/%s" % (integration_name))):
+            return json_error("The integration \"%s\" does not exist.")
+        else:
+            return json_error("The integration \"%s\" does not fixtures.")
```

Comment:
I'm not sure what the purpose of this `json_error` is. Could you please explain under what scenario the control would fall to this `else` block?

Reply:
It's for if someone forgets to make a 'fixtures' directory for their integration. Though now that I look at it again, the extra `if` check is a bit paranoid, I'll be removing it.